src/main/java/javaslang/collection/List.java

```diff
@@ -133,13 +136,19 @@
             @SuppressWarnings("unchecked")
             final List<T> list = (List<T>) elements;
             return list;
-        } else if (elements instanceof ArrayList || elements instanceof Vector) {
-            @SuppressWarnings("unchecked")
-            final java.util.List<T> indexedList = (java.util.List<T>) elements;
+        } else if (elements instanceof java.util.List<?>) {
```

Comment:
No, there is no advantage as far as I can tell. It's probably just me: when I see a generic type reference without any type parametrization I feel inclined to add that.